### PR TITLE
Add python3 compatibility

### PR DESCRIPTION
**Pull request by j-licht**
To use anymore with python2 you must depend on the future package.

### CODE REVIEW

File: src/check_ceph_osd [hunk 1]
```diff
@@ -106,9 +107,10 @@ def main():
   # exec command
   p = subprocess.Popen(ceph_cmd,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
   output, err = p.communicate()
+  output = output.decode('utf8')
```
Comment (valerytschopp) [inline src/check_ceph_osd:110]: why is the `output.decode('utf8')` needed here? Is it python2 specific?

File: src/check_ceph_mds [hunk 2]
```diff
@@ -106,27 +109,27 @@ def check_target_mds(mds_stat, fs_name, name):
     # find mds from standby list
     standby_mdss = _get_standby_mds(mds_stat)
     for mds in standby_mdss:
-        if mds.get_name() == name:
-            print "MDS OK: %s" % (mds)
+        if mds.get_name() == name.encode('utf8'):
```
Comment (valerytschopp) [inline src/check_ceph_mds:112]: Why `name.encode('utf8')` ?

File: src/check_ceph_mds [hunk 3]
```diff
@@ -106,27 +109,27 @@ def check_target_mds(mds_stat, fs_name, name):
...
+            print("MDS OK: %s" % (mds))
             return STATUS_OK
 
     # find mds from active list
     active_mdss = _get_active_mds(mds_stat, fs_name)
 
     if active_mdss:
         for mds in active_mdss:
-            if mds.get_name() != name:
+            if mds.get_name() != name.encode('utf8'):
```
Comment (valerytschopp) [inline src/check_ceph_mds:121]: Same comment, why `name.encode('utf8')` ?

**Review by valerytschopp — COMMENTED** [review]
I'm always very careful with utf8 encoding/decoding in python2. 
Python3 does it differently, it is why I'm asking to clarify those string encode('utf8') or decode('utf8').

File: src/check_ceph_mds (hunk 3 above)
Comment (j-licht) [reply src/check_ceph_mds:121]: In the function _get_standby_mds https://github.com/ceph/ceph-nagios-plugins/blob/ec3800bfdd24c56896fdb8579c3435a44efe2740/src/check_ceph_mds#L135 the mds name is encoded to utf-8.
So the name form the args must be also encoded as utf-8 that the name matches furthermore.

File: src/check_ceph_osd (hunk 1 above)
Comment (j-licht) [reply src/check_ceph_osd:110]: Without this I got the error:
```Traceback (most recent call last):
  File "./src/check_ceph_osd", line 154, in <module>
    sys.exit(main())
  File "./src/check_ceph_osd", line 121, in main
    up = re.findall(r"^(osd\.%s) up.*%s:" % (args.osdid, osd_host), output, re.MULTILINE)
  File "/usr/lib/python3.7/re.py", line 225, in findall
    return _compile(pattern, flags).findall(string)
TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by valerytschopp** [comment]
Hi @j-licht,
Please tell me when you are finished, and I will do the last review before merging it
Cheers

**Comment by j-licht** [comment]
@valerytschopp I think I'm finished.
I currently can not test python2 support because the most required packages was already removed on my distro.

**Comment by valerytschopp** [comment]
I'm running some basic functionality tests with the following:

- Ceph version 14.2.15 (afdd217ae5fb1ed3f60e16bd62357ca58cc650e5) nautilus (stable) 
- Ubuntu 18.04.2 LTS
- Python 2.7.17
- Python 3.6.9

I've checked out: https://github.com/j-licht/ceph-nagios-plugins.git @ master

1. python 2.7

```
:~/ceph-nagios-plugins/src$ python check_ceph_mds -n s0015 -f cephfs
Traceback (most recent call last):
  File "check_ceph_mds", line 21, in <module>
    from builtins import range
ImportError: No module named builtins
```

The `from builtins import` are python3 only?!? I removed the 2 lines.

2. python 3 (after removing the builtins import)

```
:~/ceph-nagios-plugins/src$ python3 check_ceph_mds -n s0015 -f cephfs
MDS OK: MDS 'b's0015'' is b'up:active'
```

Here, the `b'` (byte prefix) of the strings is very strange!?! So I removed all the `decode('utf8')` and `encode('utf8')` from the source code:

```diff
diff --git a/src/check_ceph_mds b/src/check_ceph_mds
index 83dbefe..4e654c0 100755
--- a/src/check_ceph_mds
+++ b/src/check_ceph_mds
@@ -18,8 +18,6 @@
 
 
 from __future__ import print_function
-from builtins import range
-from builtins import object
 import argparse
 import socket
 import os
@@ -109,7 +107,7 @@ def check_target_mds(mds_stat, fs_name, name):
     # find mds from standby list
     standby_mdss = _get_standby_mds(mds_stat)
     for mds in standby_mdss:
-        if mds.get_name() == name.encode('utf8'):
+        if mds.get_name() == name:
             print("MDS OK: %s" % (mds))
             return STATUS_OK
 
@@ -118,7 +116,7 @@ def check_target_mds(mds_stat, fs_name, name):
 
     if active_mdss:
         for mds in active_mdss:
-            if mds.get_name() != name.encode('utf8'):
+            if mds.get_name() != name:
                 continue
             # target mds in active list
             print("MDS %s: %s" % ("WARN" if mds.is_laggy() else "OK", mds))
@@ -135,8 +133,8 @@ def check_target_mds(mds_stat, fs_name, name):
 def _get_standby_mds(mds_stat):
     mds_array = []
     for mds in mds_stat['fsmap']['standbys']:
-        name = mds['name'].encode('utf8')
-        state = mds['state'].encode('utf8')
+        name = mds['name']
+        state = mds['state']
         laggy_since = mds['laggy_since'] if 'laggy_since' in mds else None
         mds_array.append(MDS(name, state))
 
@@ -154,8 +152,8 @@ def _get_active_mds(mds_stat, fs_name):
         mds_array = []
         infos = mds_stat['fsmap']['filesystems'][i]['mdsmap']['info']
         for gid in infos:
-            name = infos[gid]['name'].encode('utf8')
-            state = infos[gid]['state'].encode('utf8')
+            name = infos[gid]['name']
+            state = infos[gid]['state']
             laggy_since = infos[gid]['laggy_since'] if 'laggy_since' in infos[gid] else None
             mds_array.append(MDS(name, state, laggy_since))
 
```

And now both python2 and python3 works, and give the same output:

1. python 2.7:
```
tschopp@s0007:~/ceph-nagios-plugins/src$ python check_ceph_mds -n s0015 -f cephfs
MDS OK: MDS 's0015' is up:active
```

2. python 3
```
tschopp@s0007:~/ceph-nagios-plugins/src$ python3 check_ceph_mds -n s0015 -f cephfs
MDS OK: MDS 's0015' is up:active
```

**Review by valerytschopp — CHANGES_REQUESTED** [review]
Other lines must be corrected. Basically every line with `.encode('utf8')` must be updated to remove the `encode()`.

Lines:
```
        name = mds['name'].encode('utf8')
        state = mds['state'].encode('utf8')
```

and 

```
            name = infos[gid]['name'].encode('utf8')
            state = infos[gid]['state'].encode('utf8')
```

I'm pretty sure that line 101: `        mds_stat = json.loads(output)
` deals coorectly with the encoding, decoding

**Review by valerytschopp — APPROVED** [review]

**Comment by valerytschopp** [comment]
@j-licht thanks a lot for python3 compatibility update

**Pull request merged**